plutus-metatheory/src/Algorithmic/CEK.lagda.md

```diff
@@ -360,15 +399,28 @@ step (s ; ρ ▻ unwrap L refl) = (s , unwrap-) ; ρ ▻ L
 step (s ; ρ ▻ con c refl) = s ◅ V-con c
 step (s ; ρ ▻ (builtin b / refl)) = s ◅ ival b
 step (s ; ρ ▻ error A) = ◆ A
+step (s ; ρ ▻ constr e A refl as) with Vec.lookup A e in eq 
+step (s ; ρ ▻ constr e A refl []) | []  = s ◅ V-constr e A [] (cong ([] <><_) (sym eq))
+step (_;_▻_ {Γ} s ρ (constr e A refl (_∷_ {xty} {xsty} x xs))) | _ ∷ _ = (s , constr- e A ρ (sym eq) {start} ( _ ∶ [] ∣ xs)) ; ρ ▻ x
+step (s ; ρ ▻ case t ts) = (s , case- ρ ts) ; ρ ▻ t
 step (ε ◅ V) = □ V
 step ((s , -· M ρ') ◅ V) = (s , V ·-) ; ρ' ▻ M
+step ((s , -·v v) ◅ vf)  = (s , vf ·-) ◅ v
 step ((s , (V-ƛ M ρ ·-)) ◅ V) = s ; ρ ∷ V ▻ M
 step ((s , (V-I⇒ b {am = 0} bapp ·-)) ◅ V) = s ; [] ▻ (BUILTIN' b (bapp $ V))
 step ((s , (V-I⇒ b {am = suc _} bapp ·-)) ◅ V) = s ◅ V-I b (bapp $ V)
 step ((s , -·⋆ A) ◅ V-Λ M ρ) = s ; ρ ▻ (M [ A ]⋆)
 step ((s , -·⋆ A) ◅ V-IΠ b {σB = σ} bapp) = s ◅ V-I b (_$$_ bapp refl {σ [ A ]SigTy})
 step ((s , wrap-) ◅ V) = s ◅ V-wrap V
 step ((s , unwrap-) ◅ V-wrap V) = s ◅ V
+step ((s , constr- i TSS ρ refl {tidx} chip) ◅ v) 
+    with Vec.lookup TSS i in eq
+... | [] with no-empty-≣-<>> tidx
+... | ()
+step ((s , constr- i TSS ρ refl (r ∶ vs ∣ [])) ◅ v) | A ∷ AS  = s ◅ V-constr i TSS (vs :< v) 
+                 (sym (trans (cong ([] <><_) (trans eq (lem-≣-<>> r))) (lemma<>2 _ (_ ∷ []))))
+step ((s , constr- i TSS ρ refl (r ∶ vs ∣ (t ∷ ts))) ◅ v) | A ∷ AS = (s , constr- i TSS ρ (sym eq) ((bubble r ∶ (vs :< v) ∣ ts))) ; ρ ▻ t 
+step {t} ((s , case- ρ cases) ◅ V-constr e TSS vs refl) = pushValueFrames s vs (lemma-bwdfwdfunction' (Vec.lookup TSS e)) ; ρ ▻ (lookupCase e cases) 
```

Comment:
will `vs` be big? I am wondering about the cost of `pushValueFrame`.

Reply:
At most it will be the number of arguments to `constr`.